### ISSUE TITLE
The inheritance principle is not completely supported

### ISSUE THREAD

**Issue by tsalo** [PennLINC/qsiprep#685]
I haven't done anything near a complete inventory of where metadata is loaded in QSIPrep, but in at least one location ([`get_metadata_for_nifti`](https://github.com/PennLINC/qsiprep/blob/527ef590f1c2bd77c8172cb9bf5e0f4cc09ea7b9/qsiprep/interfaces/bids.py#L457)), inherited metadata is not support.

I think the best solution is to use a `BIDSLayout` to load metadata instead of directly finding the (potentially multiple) JSONs associated with a NIFTI. This might be easier if we implement the Nipreps `Config` object.

**Comment by tsalo** [comment]
Found another case, where bval and bvec files are grabbed based on the DWI filename.

https://github.com/PennLINC/qsiprep/blob/527ef590f1c2bd77c8172cb9bf5e0f4cc09ea7b9/qsiprep/workflows/dwi/merge.py#L198-L204

**Comment by akimbler** [comment]
Partially related to this, I've noticed that when running `qsirecon`. I frequently get outputs that seem to be jammed together combinations of other input filenames with updated entities appended, rather than proper inheritance via entities. 
![image](https://github.com/PennLINC/qsiprep/assets/21314171/589457ee-59cb-4129-a485-6c38adc8976f)
![image](https://github.com/PennLINC/qsiprep/assets/21314171/6d8141e5-6ed3-4190-afed-eda9626e28e0)

**Comment by mattcieslak** [comment]
I agree, the qsirecon output names need a major overhaul. That's underway in #688 and will be in the next release. We're going to keep it close to BEP016

**Comment by tsalo** [comment]
#760 removes some of the JSON-loading functions, but doesn't deal with `read_nifti_sidecar`, which is a big barrier to using the inheritance principle in DWI datasets.
